packages/api/src/decorator/private/templateMiddleware.ts

```diff
@@ -20,7 +20,7 @@ export default function templateMiddleware<Init, Request = any, Props extends {}
   const warnInvalid = warnOnce(`"${name}" prop is invalid`);
 
   const initMiddleware = (
-    middleware: readonly MiddlewareWithInit<Middleware, Init>[],
+    middleware: readonly MiddlewareWithInit<ComponentMiddleware<any, any>, any>[],
```

Comment:
Here we expect any Middleware, but we'll get the list of ones which are typed appropriately.